### DISCUSSION TITLE
Connected to my relay server with different public keys

### DISCUSSION THREAD

**Discussion by gezusk** [rustdesk/rustdesk#4610] [Q&A]
I had set up a relay server on Oracle, but when my trial ended, it was shut down. So, I set up a second one on Google, with a new installation. When I finished, I added the info, and the new public key to my client, and it worked. Then, I saw all of my other clients also appearing online, even though they still have the previous server's public key. Is that supposed to happen? Does it break anything if I leave them as is?

**Comment by gezusk** [comment]
Also, I expected that my client IDs would change, since it was a new server, but all of them stayed the same. Are IDs unique globally?

**Comment by SergeyMy** [comment]
client ID is derived from MAC addresses :)

**Comment by gezusk** [comment]
That makes sense. But I'm still wondering about the first question. I thought having a key prevents other clients from connecting to the server that don't have the key.

**Comment by SergeyMy** [comment]
https://rustdesk.com/docs/en/self-host/install/

> If you want to prohibit users without the key from establishing non-encrypted connections, please add the -k _ parameter when running hbbs and hbbr
